app/services/datetime_parser.rb

```diff
@@ -30,10 +34,14 @@ def parse
   def check_date_is_valid
     day, month, year = raw_date.values_at(:day, :month, :year)
     Date.strptime("#{day}-#{month}-#{year}", "%d-%m-%Y")
+  rescue ArgumentError
+    issues << { schedule_date: "Enter a valid date" }
```

Comment:
This is nice!

Reply:
I wonder if `schedule_date` should just be `date` and `schedule_time` should just be `time` to make to make this class reusable? 
Or we could borrow from Content Publisher and [pass in an `issue_prefix`](https://github.com/alphagov/content-publisher/blob/2b7c5ddfa72d518064ae50d24c9bb3a1c7a1b3a6/app/services/datetime_parser.rb#L4)

Reply:
I removed the `issue_prefix` in the interests of only copying what's needed.
But I agree renaming to `date` and `time` would work and make it more reusable 👍